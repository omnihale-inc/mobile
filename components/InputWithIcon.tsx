import { ReactNode } from "react";
import { StyleSheet, TextInput } from "react-native";

import { View } from "./Themed";

const InputWithIcon = ({
  value,
  onChangeText,
  icon
}: {
  value?: string;
  onChangeText?: (value: string) => void;
  icon: (props?: object) => ReactNode;
}) => {
  return (
    <View style={styles.emailInputWrapper}>
      <TextInput
        style={styles.emailInput}
        placeholder="Enter Email address"
        keyboardType="email-address"
        value={value}
        onChangeText={onChangeText}
      />
      <View style={styles.emailIcon}>{icon()}</View>
    </View>
  );
};

const styles = StyleSheet.create({
  emailInputWrapper: {
    position: "relative"
  },
  emailInput: {
    borderColor: "#d1d3d9",
    fontFamily: "SatoshiRegular",
    borderWidth: 1,
    marginVertical: 24,
    paddingVertical: 15,
    paddingLeft: 55,
    paddingRight: 40,
    borderRadius: 30,
    fontSize: 17
  },
  emailIcon: { position: "absolute", top: 43, left: 22 }
});

export default InputWithIcon;
