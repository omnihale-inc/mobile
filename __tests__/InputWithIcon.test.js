import renderer from "react-test-renderer";

import InputWithIcon from "../components/InputWithIcon";

it("render input with icon", () => {
  const FakeComponent = jest.fn();
  const tree = renderer
    .create(
      <InputWithIcon
        icon={() => <FakeComponent />}
        onChangeText={jest.fn()}
        value="fake value"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
