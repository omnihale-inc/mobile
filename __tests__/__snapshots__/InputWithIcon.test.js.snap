// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render input with icon 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "position": "relative",
      },
    ]
  }
>
  <TextInput
    keyboardType="email-address"
    onChangeText={[MockFunction]}
    placeholder="Enter email address"
    style={
      {
        "borderColor": "#d1d3d9",
        "borderRadius": 30,
        "borderWidth": 1,
        "fontFamily": "SatoshiRegular",
        "fontSize": 17,
        "marginVertical": 24,
        "paddingLeft": 55,
        "paddingRight": 40,
        "paddingVertical": 15,
      }
    }
    value="fake value"
  />
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "left": 22,
          "position": "absolute",
          "top": 43,
        },
      ]
    }
  />
</View>
`;
